docs/connecting-data/connecting-to-apis-and-external-data/clickup-api.md

```diff
@@ -1,7 +1,186 @@
+
+Once you have filled out all the fields, press the green 'Save' button, and your API will be connected!
+
+<figure markdown>
+![ClickUp form][image-3]{width=500}
+</figure>
+
+### Name
+Enter any name that will help you recognize this data source from within your query window.
+
+!!! info "Acceptable Characters Include"
+
+    - lowercase alphanumeric characters
+    - underscores
+
+### API Key
+An API key is generated within your account page. The following steps will navigate you to its location. Once created, copy the key and enter it in the ClickUp form under 'API key'.
+
+=== "1. Home page"
+    In the ClickUp app, click your profile icon in the bottom left corner of the page.
+
+    ![Click on your ClickUp account profile icon.][image-4]
+
+=== "2. Apps"
+    Choose **Apps**.
+
+    ![Choose 'Apps'.][image-5]
+
+=== "3. API Token"
+    In the API Token section, click **Copy** (or **Regenerate** to get a new API token and then **Copy**) to save the API token to your clipboard.
+
+    ![Click 'Copy' to save API token to clipboard.][image-6]
+
+## Endpoints
+Please see [ClickUp's API Reference](https://clickup.com/api){target=_blank} for more on ClickUp's endpoints.
+
+The table below shows a list of endpoints available to connect to within the DataDistillr application. If you need to connect to any endpoints not listed in the table below, please use the [Custom APIs](https://docs.datadistillr.com/connecting-data/connecting-to-apis-and-external-data/custom-apis/) Form.
+
+| Endpoint | Required | Optional | Description |
+|  ----------- | ----------- | ----------- | ----------- |
+| `/list/{list_id}/task` | | archived<br>page<br>order_by<br>reverse<br>subtasks<br>statuses<br>include_closed<br>assignees<br>due_date_gt<br>due_date_lt<br>due_created_gt<br>due_created_lt<br>due_updated_gt<br>due_updated_lt<br>custom_fields | Returns all of the tasks associated with the given list. |
+| `/team/{team_id}/space` | | archived | Returns all of the spaces associated with the given team. |
+| `/space/{space_id}/folder` | | archived | Returns all of the folders within the given space. |
+| `/folder/{folder_id}/list` | | archived | Returns all of the lists within the given folder. |
+| `/space/{space_id}/list` | | archived | Returns all of the lists within the given space. |
+| `/space/{space_id}/tag` | | archived | Returns all of the tags associated with the given space. |
+
+### Nav Tree
+
+The endpoints above will display as follows in the nav tree once your API has successfully connected.
+
+<figure markdown>
+  ![ClickUp nav tree][image-7]{width=500}
+</figure>
+
+## Sample Queries
+
+The following queries are intended to help you get started, and make life simpler querying within your API.
+
+For the following examples, suppose that my ClickUp API data source was called `clickupapi2000`, and I want to query an endpoint. The endpoint goes after the ClickUp data source name:
```

Comment:
Does the data source need back ticks in ClickUp? In the top there are back ticks, in the samples there are not

Reply:
I'm not sure about that. I would assume back ticks are necessary because I think this is a DDR thing, and there are back ticks in the JIRA one.